doc/src/contracts/voting_verifier.md

```diff
@@ -16,29 +18,69 @@ subgraph Axelar
     V{"Verifier"}
     R{"Service Registry"}
 end
-OC{"Off-Chain Procceses"}
+OC{"Workers"}
 
 G--"VerifyMessages([M, M', M''])"-->V
 V--"VerifyMessages([M, M', M''])"-->Vr
-OC --"StartPoll([M, M', M''])"-->Vr
 Vr--"GetActiveWorkers"-->R
 OC--"Vote(poll_id, votes)"-->Vr
 OC--"EndPoll(poll_id)"-->Vr
 Vr--"MessagesVerified([M,M',M''])"-->V
 
 ```
 
+## Code & Entity Structure
+To better understand the entities and functionalities, we suggest the following order to read the `voting-verifier` contract code:
+
+1. **[state.rs](https://github.com/axelarnetwork/axelar-amplifier/blob/main/contracts/voting-verifier/src/state.rs)**
+2. **[msg.rs](https://github.com/axelarnetwork/axelar-amplifier/blob/main/contracts/voting-verifier/src/msg.rs)**
+3. **[query.rs](https://github.com/axelarnetwork/axelar-amplifier/blob/main/contracts/voting-verifier/src/query.rs)**
+4. **[execute.rs](https://github.com/axelarnetwork/axelar-amplifier/blob/main/contracts/voting-verifier/src/execute.rs)**
+5. **[contracts.rs](https://github.com/axelarnetwork/axelar-amplifier/blob/main/contracts/voting-verifier/src/contract.rs)**
+
+There are two types of `Poll`: 
+
+1. The first one is defined in `state.rs`:
+    ```Rust
+    pub enum Poll {
+        Messages(WeightedPoll),
+        ConfirmWorkerSet(WeightedPoll),
+    }
+    ```
+    When dealing with gateways deployed on external chains, 
+    there is a requirement to inform them about the workers from whom they can accept signatures. While the `service-registry` 
+    manages this process for the Axelar chain, an alternative mechanism is needed for external chains.
+    
+    The internal `Poll` enum handles this need. It is employed whenever there is a necessity to update the set of workers—whether it involves adding a 
+    new worker or removing an existing one. In such cases, a message is generated, and this message must be relayed to the external gateway. It is also 
+    important to confirm that the external chain has received the message to finalize the process.
+
+
+2. The second one resides in `msg.rs`:
+    ```Rust
+   pub struct Poll {
+        poll_id: PollId,
+        messages: Vec<Message>,
+    }
+   ```
+
+
+
+
+
```

Comment:
I don't think comments about the code should be in this doc. I think this doc should be higher level, and the only code mentioned here should be code from the API. How it's actually implemented I think shouldn't be described here. I think comments like this are more appropriate for the source code.

Reply:
Fixed here: https://github.com/axelarnetwork/axelar-amplifier/pull/237/commits/f0695dcaf709ab3c1327043adea7271fc7a10707